src/control/cmd/daos_agent/mgmt_rpc.go

```diff
@@ -182,15 +191,31 @@ func (mod *mgmtModule) handleGetAttachInfo(ctx context.Context, reqb []byte, pid
 		return nil, err
 	}
 
-	mod.monitor.RegisterProcess(ctx, pid)
-
 	return cacheResp, err
 }
 
+func (mod *mgmtModule) handlePoolConnect(ctx context.Context, reqb []byte, pid int32) ([]byte, error) {
```

Comment:
This function is always returning (nil, *).  Since you are not using the []byte return data, it would make sense to remove it, or change the implementation around if you intended to return something useful here.

Reply:
Changed the prototypes.